### PR TITLE
[office-js-preview] (Outlook) Add Feb-Mar preview features

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by ElizabethSamuel-MSFT** [comment]
This is pending review so please do not merge yet. Thanks.

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -9573,6 +9573,124 @@ declare namespace Office {
...
+         * 
+         * @beta
+         */
+        enum CategoryColor {
+            /**
+             * Default color
+             */
+            None,
+            /**
+             * Red
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Would it be possible to add the hex value to these colors? Right now, the comments add little value.

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -11875,6 +11993,21 @@ declare namespace Office {
          * </table>
          */
         body: Body;
+        /**
+         * Gets an object that provides methods for managing the item's categories.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Consider "Gets an object that manages the item's categories."

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -9573,6 +9573,124 @@ declare namespace Office {
              */
             Removed = "removed"
         }
+        /**
+         * Specifies the category color.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Consider clarifying what "category" means here.

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -9573,6 +9573,124 @@ declare namespace Office {
...
+             */
+            Blue,
+            /**
+             * Purple
+             */
+            Purple,
+            /**
+             * Peach/Brown
+             */
+            Peach, // TODO: Confirm if this or Brown
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Let's track the TODOs internally.

File: types/office-js-preview/index.d.ts [hunk 5]
```diff
@@ -12746,6 +12879,21 @@ declare namespace Office {
          * </table>
          */
         body: Body;
+        /**
+         * Gets an object that provides methods for managing the item's categories.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Same as above.

File: types/office-js-preview/index.d.ts [hunk 6]
```diff
@@ -13515,6 +13663,21 @@ declare namespace Office {
...
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Same as above.

File: types/office-js-preview/index.d.ts [hunk 7]
```diff
@@ -14898,6 +15061,21 @@ declare namespace Office {
...
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Same as above.

File: types/office-js-preview/index.d.ts [hunk 8]
```diff
@@ -15739,6 +15917,21 @@ declare namespace Office {
...
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Same as above.

File: types/office-js-preview/index.d.ts [hunk 9]
```diff
@@ -17200,6 +17393,122 @@ declare namespace Office {
         removeHandlerAsync(eventType: Office.EventType, callback?: (asyncResult: Office.AsyncResult<void>) => void): void;
     }
 
+    /**
+     * Represents the categories on an item.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Let's clarify what "categories" are a little bit more.

File: types/office-js-preview/index.d.ts [hunk 10]
```diff
@@ -17200,6 +17393,122 @@ declare namespace Office {
...
+         *
+         * @remarks
+         * <table>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>ReadWriteItem</td></tr>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
+         * </table>
+         * 
+         * @beta
+         */
+        addAsync(categories: string[], options?: Office.AsyncContextOptions, callback?: (asyncResult: Office.AsyncResult<void>) => void): void;
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: I think we need an overload that allows for a callback, but no options.

File: types/office-js-preview/index.d.ts [hunk 11]
```diff
@@ -17200,6 +17393,122 @@ declare namespace Office {
...
+         *
+         * @remarks
+         * <table>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>ReadWriteItem</td></tr>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
+         * </table>
+         * 
+         * @beta
+         */
+        removeAsync(categories: string[], options?: Office.AsyncContextOptions, callback?: (asyncResult: Office.AsyncResult<void>) => void): void;
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: I think we need an overload that allows for a callback, but no options.

File: types/office-js-preview/index.d.ts [hunk 12]
```diff
@@ -17200,6 +17393,122 @@ declare namespace Office {
...
+    }
+
+    /**
+     * Represents the details of an item's category.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Let's elaborate the "details" in the class description.

File: types/office-js-preview/index.d.ts [hunk 13]
```diff
@@ -17200,6 +17393,122 @@ declare namespace Office {
...
+     * <table>
+     *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>ReadItem</td></tr>
+     *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
+     * </table>
+     * 
+     * @beta
+     */
+    interface CategoryDetails {
+        /**
+         * The name of the category. Maximum length is 255 characters.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Consider tweaking the second sentence to be a complete sentence (like, "A name's maximum length...").

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
Looks good overall. Please consider these improvements.

File: types/office-js-preview/index.d.ts (hunk 4 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: I've requested review from Engineering to provide this clarification.

File: types/office-js-preview/index.d.ts [hunk 14]
```diff
@@ -9573,6 +9573,124 @@ declare namespace Office {
...
+         * [Api set: Mailbox Preview]
+         * 
+         * @remarks
+         * <table>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
+         * </table>
+         * 
+         * @beta
+         */
+        enum CategoryColor {
```
Comment (exextoc) [inline types/office-js-preview/index.d.ts]: These enum values are incorrect. We use the Presets defined by Graph here (Preset0,Preset1)
https://docs.microsoft.com/en-us/graph/api/resources/outlookcategory?view=graph-rest-1.0

File: types/office-js-preview/index.d.ts [hunk 15]
```diff
@@ -17200,6 +17393,122 @@ declare namespace Office {
...
+     * <table>
+     *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>ReadItem</td></tr>
+     *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
+     * </table>
+     * 
+     * @beta
+     */
+    interface Categories {
+        /**
+         * Adds categories to an item. Each category name must be unique to that mailbox but multiple categories can use the same color.
```
Comment (exextoc) [inline types/office-js-preview/index.d.ts]: We need to document that the categories being added need to be present in the master category list else the API will fail.

File: types/office-js-preview/index.d.ts [hunk 16]
```diff
@@ -17200,6 +17393,122 @@ declare namespace Office {
...
+         *
+         * @remarks
+         * <table>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>ReadItem</td></tr>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
+         * </table>
+         * 
+         * @beta
+         */
+        getAsync(options: Office.AsyncContextOptions, callback: (asyncResult: Office.AsyncResult<void>) => void): void;
```
Comment (exextoc) [inline types/office-js-preview/index.d.ts]: Do we not mention that the return type is a CategoryDetails object?

**Review by exextoc — CHANGES_REQUESTED** [review]

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: I don't think so. If I understand it correctly, the actual color displayed may vary by platform. @exextoc: Ankush, can you confirm this? Thanks.

File: types/office-js-preview/index.d.ts (hunk 16 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: Have updated to Office.AsyncResult<CategoryDetails[]>

File: types/office-js-preview/index.d.ts [hunk 17]
```diff
@@ -17200,6 +17363,167 @@ declare namespace Office {
...
+         * @param callback - Optional. When the method completes, the function passed in the callback parameter is called with a single parameter of 
+         *                 type Office.AsyncResult.
+         *
+         * [Api set: Mailbox Preview]
+         *
+         * @remarks
+         * <table>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>ReadWriteItem</td></tr>
+         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
+         *   <tr><td>Errors</td><td>NumberOfCategoriesExceeded - The number of categories exceeded the maximum amount.</td></tr>
```
Comment (exextoc) [inline types/office-js-preview/index.d.ts]: We haven't implemented this error

**Review by exextoc — APPROVED** [review]

File: types/office-js-preview/index.d.ts (hunk 14 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: Updated

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: Confirmed with Ankush.

File: types/office-js-preview/index.d.ts (hunk 17 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: error removed

File: types/office-js-preview/index.d.ts (hunk 10 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: added

File: types/office-js-preview/index.d.ts (hunk 11 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: added

File: types/office-js-preview/index.d.ts (hunk 12 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: added

File: types/office-js-preview/index.d.ts (hunk 4 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: Clarification provided so TODOs removed

File: types/office-js-preview/index.d.ts (hunk 15 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: Added

**Review by AlexJerabek — APPROVED** [review]
Looks good. I still have a couple unresolved suggestions at the descriptions of `categories: Categories`, which you can take or leave.

**Pull request merged**